### PR TITLE
Create teams

### PR DESCRIPTION
**Pull request by mikeymurph77**
Here I created the Team model, controller and tests.

Notes:

I'm starting to see some drawback / cases when Faker is not great inside my factories.
I tried to use a `build(:team).city_and_name` in one of my tests to assert it equalled an expected result, but when data is faked, it's nearly impossible to know the outcome. So to combat this, I created a `random_team` factory and a `team` factory so know what will be outputted for specific tests.

I haven't read about it yet but I think using a `before` in my tests would be a good practice. I tried earlier but it caused case sensitivity issues with the shoulda matcher `validate_uniqueness_of` tests.

### CODE REVIEW

**Comment by edwardloveall** [comment]
> I tried to use a build(:team).city_and_name in one of my tests to assert it equalled an expected result, but when data is faked, it's nearly impossible to know the outcome.

If you're testing the exact result of something, set up the object manually instead of using preset data, faked or not:

``` ruby
team = build(:team, name: 'Patriots', city: 'New England')

expect(team.city_and_name).to eq('New England Patriots')
```

File: spec/factories/teams.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FactoryGirl.define do
+  factory :random_team do
```
Comment (edwardloveall) [inline spec/factories/teams.rb]: I think you should delete this and modify your `:team` factory. My suggestion would be to use a [sequence](https://github.com/thoughtbot/factory_girl/blob/master/GETTING_STARTED.md#sequences) so they are unique. Your team names and cities will look something like `Patriots 17` but that's fine.

File: app/controllers/teams_controller.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+class TeamsController < ApplicationController
```
Comment (edwardloveall) [inline app/controllers/teams_controller.rb]: Why do you have a teams controller?

Comment (edwardloveall) [inline app/controllers/teams_controller.rb]: Specifically, why now? I know you'll eventually need this, but what does it have to do with this commit?

File: spec/models/team_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+describe Team do
+  subject { Team.new(name: 'Patriots', city: 'New England') }
+
+  it { should validate_presence_of(:name) }
+  it { should validate_presence_of(:city) }
+  it { should validate_uniqueness_of(:name) }
+
+  it "returns a team's full name as a string" do
+    expect(build(:team).city_and_name).to eq 'New England Patriots'
```
Comment (edwardloveall) [inline spec/models/team_spec.rb]: See my other comment: https://github.com/mikeymurph77/pickem/pull/5#issuecomment-173215145

File: app/controllers/teams_controller.rb (hunk 2 above)
Comment (mikeymurph77) [inline app/controllers/teams_controller.rb]: This is more of a habit. Whenever I create a new model, I tend to always add the controller as well. I never really thought about it, but at this stage in the app, it's just unnecessary code. 

I'll break this habit.

File: spec/models/team_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+describe Team do
+  subject { Team.new(name: 'Patriots', city: 'New England') }
+
+  it { should validate_presence_of(:name) }
+  it { should validate_presence_of(:city) }
+  it { should validate_uniqueness_of(:name) }
+
+  it "returns a team's full name as a string" do
+    team = build(:team, name: 'Patriots', city: 'New England')
```
Comment (mikeymurph77) [inline spec/models/team_spec.rb]: Very cool, I did not know you could override attributes from an existing Factory Girl factory. This is very handy for handing expectance tests like this one.

File: db/migrate/20160120011152_create_teams.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class CreateTeams < ActiveRecord::Migration
+  def change
+    create_table :teams do |t|
+      t.string :name, null: false
+      t.string :city, null: false
```
Comment (edwardloveall) [inline db/migrate/20160120011152_create_teams.rb]: I think maybe city might be the wrong word. For example New England is not a city. Perhaps location?

**Comment by mikeymurph77** [comment]
Is this good to close now?

**Comment by edwardloveall** [comment]
Yup looks good. Sorry. I don't think it alerts me of new commits, just new comments. In the future, let me know when you've made changes.

**Pull request merged**